clinica/utils/inputs.py

```diff
@@ -251,14 +257,40 @@ def clinica_file_reader(
             that do not exists in the existing file path.
 
     """
+    from multiprocessing import Manager
     from os.path import join
 
+    from joblib import Parallel, delayed
+
     from clinica.utils.exceptions import (
         ClinicaBIDSError,
         ClinicaCAPSError,
         ClinicaException,
     )
 
+    manager = Manager()
+
+    def find_sub_ses_pattern(input_directory, sub, ses, error_encountered, results):
```

Comment:
Is there a reason you didn't extract this function out of `clinica_file_reader`?
Also, I think adding a very short docstring could be useful.

Reply:
It's just that it is only used in this function to parallelize the for loop and also this is the way we did it in ClinicaDL (so I just copied the pattern).
Then there is no good reason actually ^^
We can put it outside if you prefer!

Reply:
Ok, thank you for the explanations. I'm fine with both solutions but a little bit more in favor of extracting it into a private helper function.
Extracting inner functions into private functions is usually good practice (if it doesn't use variables from the outer function's scope) as it yields a simpler code (code is less nested, functions are shorter...) and ease code reusability.